### PR TITLE
Added missing type alias and constexpr to group

### PR DESCRIPTION
**Pull request by DieGoldeneEnte**
This adds id_type, range_type, linear_id_type, as well as dimensions and fence_scope (as specified in the sycl2020 provisional specification).
To add the dimensions constexpr the template parameter 'dimensions' needed to be renamed to 'Dimensions' (uppercase 'd') to avoid a compiler error regarding shadowing (`error: declaration of 'dimensions' shadows template parameter`).

I think the datatypes used inside the file should be changed to use the new alias, but I didn't just want to change them. If you want, I can do it in the `group.hpp` file.

The tests still compile/run as before.

### CODE REVIEW

File: include/hipSYCL/sycl/libkernel/group.hpp [hunk 1]
```diff
@@ -97,15 +97,21 @@ using global_and_local_mem_fence = mem_fence<
...
 
-template <int dimensions = 1>
+template <int Dimensions = 1>
 struct group
 {
 
+  using id_type = id<Dimensions>;
+  using range_type = range<Dimensions>;
+  using linear_id_type = size_t;
+  static constexpr int dimension = Dimensions;
```
Comment (fodinabor) [inline include/hipSYCL/sycl/libkernel/group.hpp]: typo: dimension_s_

**Comment by DieGoldeneEnte** [comment]
Thanks @fodinabor!

File: include/hipSYCL/sycl/libkernel/group.hpp [hunk 2]
```diff
@@ -97,15 +97,21 @@ using global_and_local_mem_fence = mem_fence<
...
-template <int dimensions = 1>
+template <int Dimensions = 1>
 struct group
 {
 
+  using id_type = id<Dimensions>;
+  using range_type = range<Dimensions>;
+  using linear_id_type = size_t;
+  static constexpr int dimensions = Dimensions;
+  static constexpr memory_scope fence_scope = memory_scope::work_group;
```
Comment (illuhad) [inline include/hipSYCL/sycl/libkernel/group.hpp:109]: I think we should also include `memory.hpp`

Comment (DieGoldeneEnte) [reply include/hipSYCL/sycl/libkernel/group.hpp:109]: Done

**Comment by illuhad** [comment]
Thank you :)

**Pull request merged**